plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -417,6 +418,21 @@ instance HasConstantIn DefaultUni term => ReadKnownIn DefaultUni term Word8 wher
                _       -> throwing_ _EvaluationFailure
     {-# INLINE readKnown #-}
 
+-- | This allows us to transparently use values of type LiteralByteSize as
+-- built-in Integers but with a different size measure.
+instance KnownTypeAst tyname DefaultUni LiteralByteSize where
+    toTypeAst _ = toTypeAst $ Proxy @Integer
+
+-- See Note [Integral types as Integer].
```

Comment:
I think this note is wrong. It might need its own note!

Reply:
That's @effectfully 's note, so he should take a look.  Possibly I've added something that means that it no longer makes sense.

Reply:
I think Michael probably means that the Note does not apply here, not that it is wrong per se? The Note talks about how we encode integral types as `Integer` to get automatic bound checking, but here you do something very different: you need the wrapper for costing reasons and all those instances are literally the same as ones for `Integer`. I would therefore `newtype`-derive them and simply avoid referencing any Notes, `newtype`-deriving is enough of a stress that from the builtins point of view `LiteralByteSize` is indistinguishable from `Integer`.